### DISCUSSION TITLE
Week 7 - Debugging and Project Specifications

### DISCUSSION THREAD

**Discussion by coopsand** [USU-ITLS-x265-23-summer/class-repo#146] [General]
## Debugging Thoughts
It took me a minute to realize many of the mistakes that were hidden in the code we fixed. I have always struggled with debugging, so I want to continue to practice looking for mistakes. 

One of my favorite methods is to talk it out. (Rubber ducky style.) I've surprised myself many times with how effective this process can be. I'll often times stop mid-sentence and catch the mistake. If you've been sleeping on this tactic, try it out. It is super helpful.

The other debugging skill I use frequently is using `console.log()` to simply print out the values of different variables. You can identify if something has been initialized to the wrong value very easily this way.

Other note: I really loved the video we watched on the Apollo Computer Error. It goes to show that mistakes don't always present themselves clearly. Often, we still only get a word or single phrase to describe what is wrong. Even in modern day environments where they try to describe the error, they don't always have the clearest error messages.

## Project 
For my project I have decided to create a portfolio piece. My goals are listed below:
- Header: Set at the top of the page that clearly states my name, and other relevant information. (Such as email address.)
- Navigation Bar: A menu across the header that links to different pages of the website.
- About Me: Include an image (incl. alt. text) and description about myself.
- Image Portfolio: Integrated Graphic Design pieces with alternate text and image descriptions.
- UX Portfolio: Page including a description of a completed UX project and a formatted link to the prototype.
- Cohesive theme: Utilization of an external`styles.css` file to make the page have a cohesive theme.

NOTE: I would love to have a multi-paged site. Does this [video](https://www.youtube.com/watch?v=iXSSHlOe47s) provide an adequate tutorial for setting up more than one page?

## Project Question
Do any of you think I should add additional features or information to my project idea? TYIA.

**Comment by derekscholes** [comment]
I love the idea. You are correct about the video. You do it this way. If you would like some code to look at and my site, go ahead and click my links below. Feel free to open the codespace for it and you will see what I have done, just be careful not to change anything on there. Look around and see my files. In addition, I created different folders for certain things which wasn't necessary but makes it less cluttered.

[My Website](https://ginutritiontherapy.com/team.html)
[My GitHub](https://github.com/derekscholes/derekscholes.github.io)
<img width="1744" alt="Screenshot 2023-06-24 at 7 25 45 PM" src="https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/132800749/e3d33595-6bde-4b13-ad3c-596104c48b3d">

In addition, here is a screenshot of my codespace if you prefer. It shows my files and how they are integrated into my page for links.

**Comment by metasean** [comment]
> One of my favorite methods is to talk it out. (Rubber ducky style.) I've surprised myself many times with how effective this process can be. I'll often times stop mid-sentence and catch the mistake. If you've been sleeping on this tactic, try it out. It is super helpful.

I can't even begin to tell you how many times that I've started documenting a bug —so that I could ask for help— only to realize what the cause of the problem was. 

> `console.log()`

Be careful with `console.log()`.  It's great for primitive values (e.g. strings, numbers, etc.) but they can get you into a world of hurt when it comes to objects, arrays, etc. 
- https://stackoverflow.com/questions/9734487/javascript-array-logging-confusion

> I would love to have a multi-paged site.

You can have discreet pages (like the video that you shared) or you can use JS to replace different parts of a single page (or even completely replace the contents of a page).  The latter approach uses DOM manipulation, e.g. https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents

> Do any of you think I should add additional features or information to my project idea? TYIA.

At the moment, I'd focus on what you've already identified.
